models/users.js

```diff
@@ -269,6 +270,25 @@ const fetchUser = async ({ userId = null, username = null, githubUsername = null
   }
 };
 
+const fetchGithubInfoByUserId = async (userId) => {
+  if (!userId) {
+    throw new Error("User Id is missing");
+  }
+  try {
+    const userData = await fetchUser({ userId: userId });
+    const { userExists, user } = userData;
+    if (!userExists || !user.github_id) {
+      throw new Error("User not found");
+    }
```

Comment:
Can we make this funtction simpler models only need to query database, here first we are fetching user id then passing the user id to wait again for github info. make a function that fetches github info based on github id. the other part of extracting user id can be done some place else.

Reply:
Yes I had a chat with Ankush he told me to store GitHub created on value in user modal so I don't need these things
that's why I changed this PR into a draft.